### ISSUE TITLE
Select - When Inline filter field is open, allow us to also specify a valid pattern that can be searched

### ISSUE THREAD

**Issue by jeswanke** [patternfly/patternfly-react#9407]
**Is this a new component or an extension of an existing one?**
Select

**Describe the feature**
When an inline filter is open user can type any characters. even though we might know the selections are only using digits, etc.
If this happens, the component just says:
    noResultsFoundText: 'No results found',
Ideally it would say the entered text contains letters when it should just contain numbers, etc

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
maybe this?
<img width="345" alt="CleanShot 2023-07-26 at 11 13 15@2x" src="https://github.com/patternfly/patternfly-react/assets/60980397/8720ac38-0e03-4451-874b-020c6e73fa46">

**Any other information?**

**Comment by wise-king-sullyman** [comment]
You're correct that this isn't possible with the v4 Select implementation! We'll have someone investigate building a demo of this functionality using the new v5 components.

**Comment by thatblindgeye** [comment]
This seems like it would be more of a combination of using both Select and HelperText, or having the "no results found" text instead be more dynamic. @jeswanke would this be something that a quick codesandbox would be helpful? Keeping in mind PatternFly is currently on version 6 (not sure if this issue was opened against v4 or v5).
